### DISCUSSION TITLE
Need help to build IM in Linux OS

### DISCUSSION THREAD

**Discussion by VadeInPace** [ImageMagick/ImageMagick#6188] [Help]
### ImageMagick version

7.1.0-9

### Operating system, version and so on

Opensuse Leap 15.4

### Description

Hi there,

I need to build IM with the process module "srchimg".
I am not very skilled in this matter, never did a build.
The link [http://im.snibgo.com/customim.htm](url) is for Windows OS only.
I didn't find an how to for Linux, is there one?

How to install the environment and process the build for Linux?
Many thanks for your help.

**Comment by snibgo** [comment]
My instructions in [Process modules](http://im.snibgo.com/customim.htm) are for Cygwin on Windows, because that is what I use, but they should apply to any Unix-like system. (Directory names will be different, of course.)

For installing IM from source, see https://imagemagick.org/script/install-source.php#linux . I suggest you do this before re-building with my process modules. The `./configure` command should contain whatever options you need, for example `--prefix` for the directory to contain IM, and perhaps `--enable-hdri`. Check that the `analyze` process module is available: `magick -list module`. Check that required delegates (eg png, tiff etc) are available.

When your build process works, then download my process modules and re-build IM.

For my process modules, you need to download their sources from http://im.snibgo.com/imsnibgoBats.zip into the `filters\` directory. Also put `vsn_defines.h` and `imv6or7.h` in the parent directory of `filters\`. Then `filters\` will contain a load of *.c, *.h and *.inc files, and `makefile.am`. `makefile.am` contains instructions to IM for building the process modules.

Then do a total re-build of IM: `automake`, `autoconf`, `./configure`, `make` and `make install`. Check that delegates and my process modules are available.

**Reply by VadeInPace** [reply]
As I said, I am not skilled
What I did
Create de new Dir
Download IM from link
Verify with
`sha256sum ImageMagick.tar.gz`
Move ImageMagick.tar.gz in
Extract tar.gz
There is a new dir ImageMagick-7.1.1-4
From terminal
```
./configure
...
configure: error: no acceptable C compiler found in $PATH
```
I can't understand the log
Should I install the development tools module from OpenSuse repo? What exactly, there are few modules in Pattern Development;
Base Development
C/C++Development

**Comment by snibgo** [comment]
> no acceptable C compiler found in $PATH

To build IM, you need a C++ compiler and other tools such as `automake` and `autoconf` and `make`. I don't know how to get those tools on OpenSuse.
